src/bytes.rs

```diff
@@ -200,6 +201,53 @@ impl Bytes {
         }
     }
 
+    /// Create [Bytes] with a buffer whose lifetime is controlled
+    /// via an explicit owner.
+    ///
+    /// A common use case is to zero-copy construct from mapped memory.
+    ///
+    /// ```ignore
+    /// use bytes::Bytes;
+    /// use std::fs::File;
+    /// use memmap2::Map;
+    ///
+    /// let file = File::open("upload_bundle.tar.gz")?;
+    /// let mmap = unsafe { Mmap::map(&file) }?;
+    /// let b = Bytes::from_owner(mmap);
+    /// ```
+    ///
+    /// The `owner` will be transferred to the constructed [Bytes] object, which
+    /// will ensure it is dropped once all remaining clones of the constructed
+    /// object are dropped. The owner will then be responsible for dropping the
+    /// specified region of memory as part of its [Drop] implementation.
+    ///
+    /// Note that converting [Bytes] constructed from an owner into a [BytesMut]
+    /// will always create a deep copy of the buffer into newly allocated memory.
+    pub fn from_owner<T: AsRef<[u8]>>(owner: T) -> Self {
```

Comment:
We need `T: Send` because we may run the destructor of `T` on another thread.

(`T: Sync` is not needed because we don't call `as_ref` in parallel; we just call it once and stash the pointer.)

Reply:
It also needs `'static`, as demonstrated by running Miri on:


```rs
fn main() {
    let bytes = {
        let data = [0u8; 8];
        bytes::Bytes::from_owner(&data)
    };

    println!("{bytes:?}");
}
```

Reply:
Added `Send + 'static`.